js/packages/mobile-wallet-adapter-walletlib/android/src/main/java/com/solanamobile/mobilewalletadapterwalletlib/reactnative/SolanaMobileWalletAdapterWalletLibModule.kt

```diff
@@ -86,36 +95,46 @@ class SolanaMobileWalletAdapterWalletLibModule(val reactContext: ReactApplicatio
     private var scenarioUri: Uri? = null
     private var scenario: Scenario? = null
         set(value) {
-            value?.let { scenarioId = UUID.randomUUID().toString() } ?: run {
-                scenarioId = null
-                scenario?.close()
-            }
+            value?.let { scenarioId = UUID.randomUUID().toString() }
+                    ?: run {
+                        scenarioId = null
+                        scenario?.close()
+                    }
             pendingRequests.clear()
             field = value
         }
 
-    // very basic/naive implememtion of request cache. 
-    // we could replace this with an abstraction that has expiration, persistance, etc. 
+    // very basic/naive implememtion of request cache.
+    // we could replace this with an abstraction that has expiration, persistance, etc.
     private val pendingRequests = mutableMapOf<String, MobileWalletAdapterRemoteRequest>()
 
     private fun clusterToRpcUri(cluster: String?): Uri {
         return when (cluster) {
             ProtocolContract.CLUSTER_MAINNET_BETA ->
-                Uri.parse("https://api.mainnet-beta.solana.com")
-            ProtocolContract.CLUSTER_DEVNET ->
-                Uri.parse("https://api.devnet.solana.com")
-            else ->
-                Uri.parse("https://api.testnet.solana.com")
+                    Uri.parse("https://api.mainnet-beta.solana.com")
+            ProtocolContract.CLUSTER_DEVNET -> Uri.parse("https://api.devnet.solana.com")
+            else -> Uri.parse("https://api.testnet.solana.com")
         }
     }
 
     @ReactMethod
     fun createScenario(
-        walletName: String,
-        uriStr: String,
-        config: String,
+            walletName: String,
+            config: String,
     ) = launch {
-        val uri = Uri.parse(uriStr)
+        // Get the intent that started this activity
```

Comment:
This is the main change.

I've removed the `uriStr` parameter from the API. Instead we use `reactContext.getCurrentActivity()?.getIntent()` to get the Activity's intent and then feed the `intent.getData()` as the URI